connect_deprish_to_ferc1.py

```diff
@@ -181,30 +213,166 @@ def connect(inputs):
         .assign(
             level_deprish=lambda x:
                 np.where(x.part_no_deprish < x.part_no_ferc1,
-                         'beeeg', x.level_deprish),)
+                         'beeg', x.level_deprish),)
         .assign(
             level_deprish=lambda x:
                 np.where(x.part_no_deprish > x.part_no_ferc1,
                          'smol', x.level_deprish))
         .drop(columns=['part_no_deprish', 'part_no_ferc1'])
     )
+    # we are going to make a count_ferc1 column to know how many possible
+    # ferc1 connections are possible.
+    options_all_deprish_ferc1 = pd.merge(
+        options_all_deprish_ferc1,
+        options_all_deprish_ferc1.groupby(['record_id_eia_deprish'])
+        .agg({'record_id_ferc': 'count'})
+        .rename(columns={'record_id_ferc': 'count_ferc1', })
+        .reset_index(),
+    )
+    return options_all_deprish_ferc1
+
+
+def get_matches_at_diff_ownership(options_df):
+    """
+    Get matches when the record_id_eia matches except for the ownership.
+
+    Note: Is there a cleaner way to do this??
```

Comment:
You could use a regex that matches both `owned` and `total` like `(owned|total)` rather than the double-replace.  I think you could also just replace with an empty string `""` rather than using the world 'placeholder'.
```py
diff_own = (
    options_df[
        options_df.record_id_eia_deprish.str.replace('(owned|total)', "", regex=True) ==
        options_df.record_id_eia_ferc1.str.replace('(owned|total)', "", regex=True)
    ]
)
```

Reply:
thanks